### ISSUE TITLE
Assembly Manual

### ISSUE THREAD

**Issue by qyz1106** [jmontineri/OpeNITHM#8]
Is there any finished product photos or assembly manual to help in the process of assembly? I have all the required hardware printed out, but am hesitate to put them together.

Thanks
John

**Comment by leorayjes** [comment]
Bumping the thread; has anyone completed a controller and, if so, could you post more pictures of the "guts" and/or describe the assembly process?

**Comment by doov-p** [comment]
Also bumping, but looking for more of an ELI5 guide that someone with limited hardware and electronics experience could follow.

**Comment by pikapikahikaru** [comment]
Wrote a guide for the acrylic version.

Something you should check BEFORE buying materials:
**1.** The document said it needs 10 「25mm M3 Female-Female hex standoffs」and the given link is linked to a 4.5mm hex standoffs. But the standard M3 hex’s hex size is 5.5mm. So, you should check if 4.5mm hex standoffs are available in your region. If it is not available like here in Japan, you need to modify the 4.5mm hex to 5.5mm hex in the given .svg files before laser cutting the acrylic boards.
![M3](https://user-images.githubusercontent.com/66895440/107256247-97311f80-6a7c-11eb-9a20-c2a8fda665a9.jpg)
**2.** The breakout board and the air sensor board are not available in shops, you need to order them from PCB prototyping/ manufacturing services using the given gerber.zip files.
**3.** There is no enough space for 0.1" Sockets by default. If you prefer socket your Teensy and multiplexers rather than solder them to the breakout board, you may need adding TWO extra 5mm acrylic layers and it would need 35mm standoffs instead of 25mm.
**4.** Basic skills and tools for soldering are needed.
**5.** It may need extra 4 25mm M3 screws and 4 M3 hex nuts except the given parts in the list.

Getting started:
**1.**	Soldering resistors:
Solder the 33 ohm resistors to R17-R19 and the 51k ohm resistors to R20-R25 marked on the breakout board. You may also solder the angled pins to the breakout board by the way.
![resistor](https://user-images.githubusercontent.com/66895440/107256579-00189780-6a7d-11eb-9c5e-90e01e2eae0c.jpg)

**2.**	Soldering Teensy LC and USB-C breakout board
Solder the Teensy LC and the USB-C breakout board directly like this. You can also solder a 0.1" 1 row, 4 pos pin header, or a 0.1" 1 row, 4 pos socket to the back side of the Teensy LC.
![typec breakout](https://user-images.githubusercontent.com/66895440/107256704-2b02eb80-6a7d-11eb-99ca-3513fcbe0e5d.jpg)
Some USB type-c breakboards may need to be cut using something like a die grinder, depending on its size.
![IMG_6752](https://user-images.githubusercontent.com/66895440/107270364-34488400-6a8e-11eb-93cc-0ef1a7d58ed0.jpg)


**3.**	Soldering Teensy LC and multiplexers
Solder the Teensy LC and the multiplexers to the breakout board like this.
![IMG_6748](https://user-images.githubusercontent.com/66895440/107256810-4f5ec800-6a7d-11eb-898a-ea1912c39cc4.JPG)

**4.**	Soldering air sensors
Solder the IR LEDs and the phototransistors to the air sensor breakout boards like this. These boards are the two sides of **a SAME PCB**. Be cautious that **DO NOT** reverse the anode/collector and a cathode/emitter of a LED/phototransistor, or they would not work properly.
![air](https://user-images.githubusercontent.com/66895440/107261764-0e69b200-6a83-11eb-945f-e913030f9e67.jpg)
Wires should be soldered to the opposed side of the IR LEDs/phototransistors
![air back](https://user-images.githubusercontent.com/66895440/107264088-df087480-6a85-11eb-8411-c9a6f3107831.jpg)

**5.**	Soldering copper pads to a wire (Optional)
In my case, I use 0.5mm copper plates instead of copper tape. If you would like to do this, solder your copper plates to a wire before sticking them to the acrylic.

**6.**	Prepare the acrylic boards
You should have your acrylic boards (and black papers) prepared as the document told. It would need aluminum foil and glue to cover the 3mm MDF(I used the cut off part of the acrylic instead) and the black paper with foil:
![all_layers](https://user-images.githubusercontent.com/66895440/107257151-b8ded680-6a7d-11eb-9722-d1ec13f8a889.jpg)
There should be an acrylic plate looks like this:
![untitled 8](https://user-images.githubusercontent.com/66895440/107257227-cbf1a680-6a7d-11eb-9825-83450a0eb7fe.jpg)
Stick you copper tape and wires (or copper plates soldered to wires, whatever) with glues or double-sided adhesive tape to its back side **BEFORE** assembling because it would be covered by the MDF/acrylic.
![IMG_6750](https://user-images.githubusercontent.com/66895440/107260467-7a4b1b00-6a81-11eb-87fe-cb6322c32bab.JPG)

**7.**	Assembling the main body
It should be assembled by the order like this, you may need to solder the LED strip to wires first:
![untitled 4](https://user-images.githubusercontent.com/66895440/107258640-63a3c480-6a7f-11eb-9855-2d340c16fa6c.jpg)
Screws, nuts and standoff should be assembled like this:
![nuts layout](https://user-images.githubusercontent.com/66895440/107258671-6f8f8680-6a7f-11eb-88d6-befce6c1b412.jpg)
![openithm 17](https://user-images.githubusercontent.com/66895440/107258714-7a4a1b80-6a7f-11eb-9b9a-5e6c07c5e3bb.jpg)
In my case I am using 40 M3x8 flat head screws instead of 3 types of M3 hex drive screws. 

**8.**	Assembling air sensors
It should be assembled by the order like this:
![untitled 7](https://user-images.githubusercontent.com/66895440/107258870-a796c980-6a7f-11eb-85d3-a7e315e4adc7.jpg)
Screws, nuts and standoff (and PCB) should be assembled like this:
![untitled 19](https://user-images.githubusercontent.com/66895440/107258916-b8473f80-6a7f-11eb-9f73-43a029c71809.jpg)
![untitled 22](https://user-images.githubusercontent.com/66895440/107258924-bb423000-6a7f-11eb-906d-b3ba05c69966.jpg)
![untitled 20](https://user-images.githubusercontent.com/66895440/107258934-bed5b700-6a7f-11eb-84cf-6f8c4493f086.jpg)

**9.**	Combining body and air sensors
Combine them with M5x20 screws and M5 hex nuts like this:
![openithm 6](https://user-images.githubusercontent.com/66895440/107259105-f0e71900-6a7f-11eb-8e3f-55fa495866ab.jpg)
![untitled 23](https://user-images.githubusercontent.com/66895440/107259116-f6446380-6a7f-11eb-8542-eb8fe5a1ac50.jpg)

**10.**	Wiring
You should follow OpeNITHM/PCB Files/Circuit-Teensy32/OpeNITHM_Teensy_Schematic_v2.1.pdf to wire your controler properly unless you could not understand the schematic. If so, please refer this:
![wring](https://user-images.githubusercontent.com/66895440/107260672-b4b4b800-6a81-11eb-9614-4c7835991bfa.jpg)
Then it should become a mess like this:
![IMG_6747](https://user-images.githubusercontent.com/66895440/107260816-e299fc80-6a81-11eb-8e77-d1c781ddcbe0.JPG)

**11.**	Assemble the bottom acrylic plate
![openithm 16](https://user-images.githubusercontent.com/66895440/107261186-59cf9080-6a82-11eb-93cf-9c4f66f652c5.jpg)
Then follow OpeNITHM/readme.md to upload programs to the Teensy LC  and DONE.

**Comment by Shionege** [comment]
> Wrote a guide for the acrylic version.
> 
> Something you should check BEFORE buying materials:
> **1.** The document said it needs 10 「25mm M3 Female-Female hex standoffs」and the given link is linked to a 4.5mm hex standoffs. But the standard M3 hex’s hex size is 5.5mm. So, you should check if 4.5mm hex standoffs are available in your region. If it is not available like here in Japan, you need to modify the 4.5mm hex to 5.5mm hex in the given .svg files before laser cutting the acrylic boards.
> ![M3](https://user-images.githubusercontent.com/66895440/107256247-97311f80-6a7c-11eb-9a20-c2a8fda665a9.jpg)
> **2.** The breakout board and the air sensor board are not available in shops, you need to order them from PCB prototyping/ manufacturing services using the given gerber.zip files.
> **3.** There is no enough space for 0.1" Sockets by default. If you prefer socket your Teensy and multiplexers rather than solder them to the breakout board, you may need adding TWO extra 5mm acrylic layers and it would need 35mm standoffs instead of 25mm.
> **4.** Basic skills and tools for soldering are needed.
> **5.** It may need extra 4 25mm M3 screws and 4 M3 hex nuts except the given parts in the list.
> 
> Getting started:
> **1.** Soldering resistors:
> Solder the 33 ohm resistors to R17-R19 and the 51k ohm resistors to R20-R25 marked on the breakout board. You may also solder the angled pins to the breakout board by the way.
> ![resistor](https://user-images.githubusercontent.com/66895440/107256579-00189780-6a7d-11eb-9c5e-90e01e2eae0c.jpg)
> 
> **2.** Soldering Teensy LC and USB-C breakout board
> Solder the Teensy LC and the USB-C breakout board directly like this. You can also solder a 0.1" 1 row, 4 pos pin header, or a 0.1" 1 row, 4 pos socket to the back side of the Teensy LC.
> ![typec breakout](https://user-images.githubusercontent.com/66895440/107256704-2b02eb80-6a7d-11eb-99ca-3513fcbe0e5d.jpg)
> Some USB type-c breakboards may need to be cutted using something like a die grinder, depending on its size.
> ![IMG_6752](https://user-images.githubusercontent.com/66895440/107270364-34488400-6a8e-11eb-93cc-0ef1a7d58ed0.jpg)
> 
> **3.** Soldering Teensy LC and multiplexers
> Solder the Teensy LC and the multiplexers to the breakout board like this.
> ![IMG_6748](https://user-images.githubusercontent.com/66895440/107256810-4f5ec800-6a7d-11eb-898a-ea1912c39cc4.JPG)
> 
> **4.** Soldering air sensors
> Solder the IR LEDs and the phototransistors to the air sensor breakout boards like this. These boards are the two sides of **a SAME PCB**. Be cautious that **DO NOT** reverse the anode/collector and a cathode/emitter of a LED/phototransistor, or they would not work properly.
> ![air](https://user-images.githubusercontent.com/66895440/107261764-0e69b200-6a83-11eb-945f-e913030f9e67.jpg)
> Wires should be soldered to the opposed side of the IR LEDs/phototransistors
> ![air back](https://user-images.githubusercontent.com/66895440/107264088-df087480-6a85-11eb-8411-c9a6f3107831.jpg)
> 
> **5.** Soldering copper pads to a wire (Optional)
> In my case, I use 0.5mm copper plates instead of copper tape. If you would like to do this, solder your copper plates to a wire before sticking them to the acrylic.
> 
> **6.** Prepare the acrylic boards
> You should have your acrylic boards (and black papers) prepared as the document told. It would need aluminum foil and glue to cover the 3mm MDF and the black paper with foil:
> ![all_layers](https://user-images.githubusercontent.com/66895440/107257151-b8ded680-6a7d-11eb-9722-d1ec13f8a889.jpg)
> There should be an acrylic plate looks like this:
> ![untitled 8](https://user-images.githubusercontent.com/66895440/107257227-cbf1a680-6a7d-11eb-9825-83450a0eb7fe.jpg)
> Stick you copper tape and wires (or copper plates soldered to wires, whatever) with glues or double-sided adhesive tape to its back side **BEFORE** assembling because it would be covered by the MDF acrylic.
> ![IMG_6750](https://user-images.githubusercontent.com/66895440/107260467-7a4b1b00-6a81-11eb-87fe-cb6322c32bab.JPG)
> 
> **7.** Assembling the main body
> It should be assembled by the order like this, you may need to solder the LED strip to wires first:
> ![untitled 4](https://user-images.githubusercontent.com/66895440/107258640-63a3c480-6a7f-11eb-9855-2d340c16fa6c.jpg)
> Screws, nuts and standoff should be assembled like this:
> ![nuts layout](https://user-images.githubusercontent.com/66895440/107258671-6f8f8680-6a7f-11eb-88d6-befce6c1b412.jpg)
> ![openithm 17](https://user-images.githubusercontent.com/66895440/107258714-7a4a1b80-6a7f-11eb-9b9a-5e6c07c5e3bb.jpg)
> In my case I am using 40 M3x8 flat head screws instead of 3 types of M3 hex drive screws.
> 
> **8.** Assembling air sensors
> It should be assembled by the order like this:
> ![untitled 7](https://user-images.githubusercontent.com/66895440/107258870-a796c980-6a7f-11eb-85d3-a7e315e4adc7.jpg)
> Screws, nuts and standoff (and PCB) should be assembled like this:
> ![untitled 19](https://user-images.githubusercontent.com/66895440/107258916-b8473f80-6a7f-11eb-9f73-43a029c71809.jpg)
> ![untitled 22](https://user-images.githubusercontent.com/66895440/107258924-bb423000-6a7f-11eb-906d-b3ba05c69966.jpg)
> ![untitled 20](https://user-images.githubusercontent.com/66895440/107258934-bed5b700-6a7f-11eb-84cf-6f8c4493f086.jpg)
> 
> **9.** Combining body and air sensors
> Combine them with M5x20 screws and M5 hex nuts like this:
> ![openithm 6](https://user-images.githubusercontent.com/66895440/107259105-f0e71900-6a7f-11eb-8e3f-55fa495866ab.jpg)
> ![untitled 23](https://user-images.githubusercontent.com/66895440/107259116-f6446380-6a7f-11eb-8542-eb8fe5a1ac50.jpg)
> 
> **10.** Wiring
> You should follow OpeNITHM/PCB Files/Circuit-Teensy32/OpeNITHM_Teensy_Schematic_v2.1.pdf to wire your controler properly unless you could not understand the schematic. If so, please refer this:
> ![wring](https://user-images.githubusercontent.com/66895440/107260672-b4b4b800-6a81-11eb-9614-4c7835991bfa.jpg)
> Then it should become a mess like this:
> ![IMG_6747](https://user-images.githubusercontent.com/66895440/107260816-e299fc80-6a81-11eb-8e77-d1c781ddcbe0.JPG)
> 
> **11.** Assemble the bottom acrylic plate
> ![openithm 16](https://user-images.githubusercontent.com/66895440/107261186-59cf9080-6a82-11eb-93cf-9c4f66f652c5.jpg)
> Then follow OpeNITHM/readme.md to upload programs to the Teensy LC and DONE.

Just want to ask something, where can i see overall sizes of the acrylic? can't seem to fins anything about the dimensions thanks!

**Comment by vladkorotnev** [comment]
@pikapikahikaru 

> If it is not available like here in Japan, you need to modify the 4.5mm hex to 5.5mm hex in the given .svg files before laser cutting the acrylic boards.

Do you happen to still have the modified files for the proper size of the hex standoffs? If you could attach them here that would be much appreciated :-)

**Comment by pikapikahikaru** [comment]
@vladkorotnev 

> Do you happen to still have the modified files for the proper size of the hex standoffs? If you could attach them here that would be much appreciated :-)

Sure, but I only have the .cdr files for CorelDRAW 2019 or above,  which I used for laser cutting.
Maybe they can also be opened with Adobe Illustrator (not tested)
[layers_modified.zip](https://github.com/jmontineri/OpeNITHM/files/6508453/layers_modified.zip)

**Comment by vladkorotnev** [comment]
@pikapikahikaru Thank you very much! I'm also currently looking around for what company in Japan could produce such parts, but cannot find any that would work with private person, seems most only work with companies/business entities (法人). Would you mind pointing me in the right direction, which service you used to order yours?

**Comment by pikapikahikaru** [comment]
@vladkorotnev 

> Would you mind pointing me in the right direction, which service you used to order yours?

I got those acrylic boards and PCBs from Aliexpress (or taobao if you can read and type Chinese), it may take a month to ship from China. You can also try searching レンタル レーザーカッター to find if there any laser cutting service nearby.

**Comment by supahas** [comment]
Oh hey thanks for the diagram, this will come in super handy, appreciate it

**Comment by gnargle** [comment]
@pikapikahikaru if possible you should make this into a .MD and PR it into the repo (either this one or skogaby's branch). It'll be a lot more visible there than in the issues page and it looks really helpful!

**Comment by sensei-BA** [comment]
Are there all PCB picture? I need it
